src/components/WorkflowModal.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component, PropTypes} from "react";
+import request from "superagent-bluebird-promise";
+import Modal from "react-bootstrap/lib/Modal";
+import Button from "react-bootstrap/lib/Button";
+
+class WorkflowModal extends Component {
+    render() {
+        if (!this.props.workflowParams) {
+            return (
+                <Modal className="modal" show={this.props.show} bsSize="large" aria-labelledby="contained-modal-title-lg">
+                    <Modal.Title id="contained-modal-title-lg">
+                        These are not the Workflow Parameters you are looking for
+                    </Modal.Title>
+                    <Modal.Footer className="modal-footer">
+                        <Button onClick={this.props.onHide}>Close
+                        </Button>
+                    </Modal.Footer>
+                </Modal>
+            )
+        } else {
+            return (
+              <Modal className="modal" show={this.props.show} bsSize="large" aria-labelledby="contained-modal-title-lg">
```

Comment:
There's code duplication here. The modal wrapper, modal title wrapper, and modal footer are all the same. The differences are the title contents and the existence of a body. The danger of copy/pasting duplicate stuff is maintenance, specifically, someone could make a change to the modal only in one place. It can get worse if you end up with more than "if-else" ("if-elseif-elseif-elseif-elseif-else").

Collaborate has good examples of this in many places. Here's just the first I came across. The differences are tucked away in separate functions. https://github.com/levelsbeyond/cloudform-collaborate-ui/blob/9369891644ca43c5fa12559aaa408bbaf2fa0436/src/js/components/NavBar.js